tests/integration/defects.intTest.ts

```diff
@@ -0,0 +1,88 @@
+import supertest from "supertest";
+import { expect } from "chai";
+import fs from "fs";
+import path from "path";
+import { DefectsService } from '../../src/services/DefectsService';
+import { DefectsDAO } from '../../src/models/DefectsDAO';
+import { Injector } from "../../src/models/injector/Injector";
+
+const url = "http://localhost:3004/";
+const request = supertest(url)
+
+describe('defects', () => {
+    describe('getDefects', () => {
+        context('when database is populated', () => {
+            let defectsService: DefectsService;
+            const defectsData = JSON.parse(fs.readFileSync(path.resolve(__dirname, '../resources/defects.json'), 'utf8'));
+            let defectsDAO = null;
+
+            before((done) => {
+                defectsService = Injector.resolve<DefectsService>(DefectsService, [DefectsDAO]);
```

Comment:
Real DAO, so matches expected type - no need for injection here.

Reply:
agreed